addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js

```diff
@@ -124,15 +137,104 @@ class DataEgressService extends Service {
         },
       ],
     };
+    const memberAccountId = await this.getMemberAccountId(requestContext, environment.id);
 
-    const environmentScService = await this.service('environmentScService');
-    const environmentScEntity = await environmentScService.mustFind(requestContext, { id: environment.id });
-    const memberAccount = await environmentScService.getMemberAccount(requestContext, environmentScEntity);
-    await this.addEgressStoreBucketPolicy(requestContext, egressStore, memberAccount.accountId);
+    const lockService = await this.service('lockService');
+    const lockId = `bucket-policy-access-${bucketName}`;
+    await lockService.tryWriteLockAndRun({ id: lockId }, async () => {
+      await this.addEgressStoreBucketPolicy(requestContext, egressStore, memberAccountId);
+    });
 
     return egressStore;
   }
 
+  async terminateEgressStore(requestContext, environmentId) {
+    const enableEgressStore = this.settings.get(settingKeys.enableEgressStore);
+    const by = _.get(requestContext, 'principalIdentifier.uid');
+    if (!enableEgressStore) {
+      throw this.boom.forbidden('Unable to create Egress store since this feature is disabled', true);
+    }
+    const s3Service = await this.service('s3Service');
+    const workspaceId = environmentId;
+    let egressStoreScanResult = [];
+
+    try {
+      egressStoreScanResult = await this._scanner()
+        .limit(1000)
+        .scan()
+        .then(egressStores => {
+          return egressStores.filter(store => store.workspaceId === workspaceId);
+        });
+    } catch (error) {
+      throw this.boom.notFound(`Error in terminating egress store: ${JSON.stringify(error)}`, true);
+    }
+    const egressStoreInfo = egressStoreScanResult[0];
+    const egressStoreStatus = egressStoreInfo.status;
+
+    if (egressStoreStatus.toUpperCase() === PROCESSING_STATUS_CODE) {
+      throw this.boom.forbidden(
+        `Egress store: ${egressStoreInfo.id} is still in processing, workspce can not be terminated before egress request is processed.`,
+        true,
+      );
+    } else if (egressStoreStatus.toUpperCase() === PROCESSED_STATUS_CODE) {
+      // ONLY terminate the egress store if it has been processed
+
+      try {
+        await s3Service.clearPath(egressStoreInfo.s3BucketName, egressStoreInfo.s3BucketPath);
+      } catch (error) {
+        throw this.boom.badRequest(
+          `Error in deleting egress store:${egressStoreInfo.s3BucketName} in bucket: ${egressStoreInfo.s3BucketPath}`,
+          true,
+        );
+      }
+      egressStoreInfo.status = TERMINATED_STATUS_CODE;
+      egressStoreInfo.updatedBy = by;
+      egressStoreInfo.updatedAt = new Date().toISOString();
+      await runAndCatch(
+        async () => {
+          return this._updater()
+            .condition('attribute_exists(id)') // yes we need this to ensure the environment does not exist already
```

Comment:
can you clarify this comment. Did you mean egressStore instead of environment?

Reply:
Fix applied in commit: 3917dc0ed4b0fa6bb7aafbd67d45cd7846a58922